src/navigation/Nav.jsx

```diff
@@ -0,0 +1,284 @@
+export class Nav extends React.Component {
+	/**
+	 * @return {React.element} the navbar component
+	 */
+	render() {
+		const {
+			media,
+			self,
+			logoAccessible,
+			navItems,
+			dropdownLoaderLabel,
+			localeCode,
+		} = this.props;
+		const {
+			login,
+			signup,
+			proDashboard,
+			explore,
+			messages,
+			notifications,
+			groups,
+			profile,
+		} = navItems;
+		const isLoggedOut = self.status === 'prereg' || !self.name;
+
+		const showScriptLogo = Boolean(media.isAtLargeUp || isLoggedOut);
+		const showSwarmLogo = Boolean(
+			media.isAtMediumUp && !media.isAtLargeUp && !isLoggedOut
+		);
+		const notificationContent = notifications.list ? (
+			<NotificationsDropdown
+				self={self}
+				notifications={notifications.list}
+				onMarkReadAction={notifications.notificationsDropdown.markRead}
+				localeCode={localeCode}
+				emptyContentLabel={notifications.notificationsDropdown.emptyContentLabel}
+				notificationsTitleLabel={notifications.label}
+				generateClassicUrl={
+					notifications.notificationsDropdown.generateClassicUrl
+				}
+			/>
+		) : (
+			<DropdownLoader label={dropdownLoaderLabel} />
+		);
+
+		const profileContent = groups ? (
+			<ProfileDropdown
+				settings={profile.profileDropdown.settings}
+				help={profile.profileDropdown.help}
+				logout={profile.profileDropdown.logout}
+				groupHome={profile.profileDropdown.groupHome}
+				allGroupsLabel={profile.profileDropdown.allGroupsLabel}
+				profile={profile}
+				groups={groups.list}
+			/>
+		) : (
+			<DropdownLoader label={dropdownLoaderLabel} />
+		);
+
+		let unauthItems = [
+			{
+				shrink: true,
+				linkTo: login.link,
+				label: login.label,
+				className: `${CLASS_UNAUTH_ITEM} navItem--login`,
+			},
+			{
+				shrink: true,
+				onAction: () => {},
+				label: signup.label,
+				className: CLASS_UNAUTH_ITEM,
+			},
+		];
+
+		let authItems = [
+			self.isProMember && {
+				shrink: true,
+				linkTo: proDashboard.link,
+				label: proDashboard.label,
+				className: cx(
+					'text--blue text--bold navItemLink--create display--none atMedium_display--block'
+				),
+			},
+			{
+				shrink: true,
+				linkTo: explore.link,
+				label: explore.label,
+				className: cx(CLASS_AUTH_ITEM, { 'flush--left': !media.isAtMediumUp }),
+				icon: <Icon shape="search" size="s" className="atMedium_display--none" />,
+			},
+			{
+				shrink: true,
+				linkTo: groups.link,
+				label: groups.label,
+				className: `atMedium_display--none ${CLASS_AUTH_ITEM}`,
+				icon: <Icon shape="groups" size="s" className="display--block" />,
+			},
+			{
+				shrink: true,
+				linkTo: messages.link,
+				label: messages.label,
+				className: cx('navItem--messages', CLASS_AUTH_ITEM),
+				icon: (
+					<Icon
+						shape="messages"
+						size="s"
+						className="display--block atMedium_display--none"
+					/>
+				),
+				hasUpdates: messages.unreadMessages > 0,
+			},
+			{
+				shrink: true,
+				linkTo: media.isAtMediumUp ? false : notifications.link,
```

Comment:
here and below: should this technically be an empty string (or null) instead of false? to keep the type the same for this prop